core/lib/multivm/src/versions/vm_latest/oracles/storage.rs

```diff
@@ -22,14 +26,30 @@ use crate::{
         old_vm::{
             history_recorder::{
                 AppDataFrameManagerWithHistory, HashMapHistoryEvent, HistoryEnabled, HistoryMode,
-                HistoryRecorder, StorageWrapper, VectorHistoryEvent, WithHistory,
+                HistoryRecorder, StorageWrapper, TransientStorageWrapper, VectorHistoryEvent,
+                WithHistory,
             },
             oracles::OracleWithHistory,
         },
         utils::logs::StorageLogQuery,
     },
 };
 
+/// We employ the following rules for cold/warm storage rules:
+/// - We price a single "I/O" access as 2k ergs. This means that reading a single storage slot
+/// would cost 2k ergs, while writing to it would 4k ergs (since it involves both reading during execution and writing at the end of it).
+/// - Thereafter, "warm" reads cosst 30 ergs, while "warm" writes cost 60 ergs. Warm writes to account for the fact that they may be reverted
+/// and so require more RAM to store them.
+
+/// If a write is after a read, we do not take into account the cost for reading the initial value from storage.
+const COLD_WRITE_AFTER_WARM_READ_COST: u32 =
+    STORAGE_ACCESS_COLD_WRITE_COST - STORAGE_ACCESS_COLD_READ_COST;
+
+const WARM_READ_REFUND: u32 = STORAGE_ACCESS_COLD_READ_COST - STORAGE_ACCESS_WARM_READ_COST;
+const WARM_WRITE_REFUND: u32 = STORAGE_ACCESS_COLD_WRITE_COST - STORAGE_ACCESS_WARM_WRITE_COST;
+const COLD_WRITE_AFTER_WARM_READ_REFUND: u32 =
```

Comment:
This can be simplified to just `STORAGE_ACCESS_COLD_READ_COST`.

Reply:
In this case "read + write" will be cheaper than "write", since cold read cost is 2000, while cold write cost is 5500

Reply:
Sorry, updating those constants will require additioanl changes to circuits, so better not do it

Reply:
Exactly. I mean doing that does not change the constant's value, so it looks like a bug to me.